pkg/controller/node/ipam/cloud_cidr_allocator.go

```diff
@@ -207,20 +210,19 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 	}
 	podCIDR := cidr.String()
 
-	for rep := 0; rep < cidrUpdateRetries; rep++ {
-		node, err = ca.nodeLister.Get(nodeName)
-		if err != nil {
-			glog.Errorf("Failed while getting node %v to retry updating Node.Spec.PodCIDR: %v", nodeName, err)
-			continue
-		}
+	node, err = ca.nodeLister.Get(nodeName)
+	if err != nil {
```

Comment:
I still don't agree this change is what we want. In most of cases, retry back-to-back will actually solve the problem with conflict. I really think that those retries here are actually something we want.

Reply:
So to clarify - I think this part should actually be reverted in my opinion. I think we want to retry-back to back (maybe not 5 times), because in majority of cases, second retry will solve the problem.

Reply:
Discussed this offline... I'm keeping this change of requeuing in case `updateCIDRAllocation` returned with an error. In general, the pattern we're following for controllers is to requeue the workitem for processing instead of putting the retry logic within the processing function itself (ref point #8 of https://github.com/kubernetes/community/blob/master/contributors/devel/controllers.md#guidelines). This is needed to be fair with other items in the queue. For e.g for endpoints controller service updates are happening like this (https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/endpoint/endpoints_controller.go#L136-L141).

That said, we're making a special exception here for the node Patch() operation (by retrying it few times within the function itself) as we've already computed the diff and it's not going to change later.. So we can save some effort.

> in majority of cases, second retry will solve the problem

I've changed the retry count to 3 just to be safe.